rx/core/observable/observable.py

```diff
@@ -208,8 +215,67 @@ def set_disposable(_: abc.Scheduler = None, __: Any = None):
         # Hide the identity of the auto detach observer
         return Disposable(auto_detach_observer.dispose)
 
-
-    def pipe(self, *operators: Callable[['Observable'], 'Observable']) -> 'Observable':
+    @overload
+    def pipe(self) -> 'Observable':  # pylint: disable=no-self-use
+        ... # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self, op1: Callable[['Observable'], A]) -> A:  # pylint: disable=function-redefined, no-self-use
+        ... # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B]) -> B:  # pylint: disable=function-redefined, no-self-use
+        ... # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B],
+             op3: Callable[[B], C]) -> C:  # pylint: disable=function-redefined, no-self-use
+        ... # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B],
+             op3: Callable[[B], C],
+             op4: Callable[[C], D]) -> D:  # pylint: disable=function-redefined, no-self-use
+        ...  # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B],
+             op3: Callable[[B], C],
+             op4: Callable[[C], D],
+             op5: Callable[[D], E]) -> E:  # pylint: disable=function-redefined, no-self-use
+        ...  # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B],
+             op3: Callable[[B], C],
+             op4: Callable[[C], D],
+             op5: Callable[[D], E],
+             op6: Callable[[E], F]) -> F:  # pylint: disable=function-redefined, no-self-use
+        ...  # pylint: disable=pointless-statement
+
+    @overload
+    def pipe(self,
+             op1: Callable[['Observable'], A],
+             op2: Callable[[A], B],
+             op3: Callable[[B], C],
+             op4: Callable[[C], D],
+             op5: Callable[[D], E],
+             op6: Callable[[E], F],
+             op7: Callable[[E], G]) -> G:  # pylint: disable=function-redefined, no-self-use
```

Comment:
I believe it should be `F` instead of `E`, i.e. 
```python
op6: Callable[[E], F],
op7: Callable[[F], G]) -> G:
```

Reply:
Thanks for spotting the typo (since there's really no help in the typing system). Go figure ....